app/models/manageiq/providers/vmware/inventory/collector/network_manager.rb

```diff
@@ -0,0 +1,152 @@
+class ManageIQ::Providers::Vmware::Inventory::Collector::NetworkManager < ManageIQ::Providers::Vmware::Inventory::Collector
+  attr_reader :network_name_mapping
+  VappNetwork = Struct.new(:id, :name, :type, :is_shared, :gateway, :dns1, :dns2, :netmask, :enabled, :dhcp_enabled)
+
+  def orgs
+    return @orgs if @orgs.any?
+    @orgs = connection.organizations
+  end
+
+  def vdc_networks
+    return @vdc_networks if @vdc_networks.any?
```

Comment:
hm, do we want to memoize it even when @vdc_networks == [] ?

We usually do just:
```ruby
def vdc_networks
    @vdc_networks ||= orgs.each_with_object([]) do |org, res|
      res.concat(org.networks.all)
   end
end
```

Reply:
I think yes :) If there were no networks and empty list was returned from API it makes no sense to perform another request during the same refresh - because empty list will be returned again. I think we would only spend more time and produce clutter HTTP traffic.

Or?